src/textual/widgets/_datatable.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import ClassVar, Generic, TypeVar, cast
+
+from rich.padding import Padding
+from rich.segment import Segment
+from rich.style import Style
+from rich.text import Text, TextType
+
+from .._lru_cache import LRUCache
+from .._segment_tools import line_crop
+from .._types import Lines
+from ..geometry import Size
+from ..reactive import Reactive
+from ..scroll_view import ScrollView
+from ..widget import Widget
+
+CellType = TypeVar("CellType")
+
+
+@dataclass
+class Column:
+    label: Text
+    width: int
+    visible: bool = False
+    index: int = 0
+
+
+@dataclass
+class Cell:
+    value: object
+
+
+class Header(Widget):
+    pass
+
+
+class DataTable(ScrollView, Generic[CellType]):
+
+    CSS = """
+    DataTable {
+        background: $surface;
+        color: $text-surface;       
+    }
+    DataTable > .datatable--header {        
+        text-style: bold;
+        background: $primary;
+        color: $text-primary;
+    }
+    DataTable > .datatable--fixed {
+        text-style: bold;
+        background: $primary-darken-2;
+        color: $text-primary-darken-2;
+    }
+
+    DataTable > .datatable--odd-row {
+        
+    }
+
+    DataTable > .datatable--even-row {
+        background: $primary 10%;
+    }
+
+    DataTable > .datatable--highlight {
+        background: $secondary;
+        color: $text-secondary;
+    }
+    """
+
+    COMPONENT_CLASSES: ClassVar[set[str]] = {
+        "datatable--header",
+        "datatable--fixed",
+        "datatable--odd-row",
+        "datatable--even-row",
+        "datatable--highlight",
+    }
+
+    def __init__(
+        self,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+    ) -> None:
+        super().__init__(name=name, id=id, classes=classes)
+        self.columns: list[Column] = []
+        self.data: dict[int, list[CellType]] = {}
```

Comment:
Do you have any plans for the integration/support of `rich.table` objects? E.g. in a setting where some other code already uses rich and produces a `Table`, having the option of using that object directly and display it within a textual app would be amazing.
I am not sure whether such a tight integration is possible/planned/desired though :thinking:

Reply:
Any Rich renderable may be used in Textual, including Tables.

Reply:
I meant a closer relation between a `rich.table` and a `DataTable`. I.e. that textual would be able to treat the former identical to (or at least to some degree like) the latter.